lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +855,41 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Check if the input of this method (vector of NodeValues) can be 
+/// legal input of CreateConcat function.
+/// \returns the dimension where the Concat will take place. If not 
+/// concat-able, return -1.
+size_t checkInputsForConcat(std::vector<NodeValue> inputs) {
+  auto firstDims = inputs.front().dims();
+  auto dimSize = firstDims.size();
+  auto index = -1;
+  auto count = 0;
+  for (auto &IN : inputs) {
+    auto curDims = IN.dims();
+    // All the inputs should have same dimension size.
+    if (curDims.size() != dimSize) {
+      return -1;
+    }
+    count = 0;
+    index = -1;
+    for (size_t i = 0; i < dimSize; ++i) {
+      // Find the index i where firstDims[i] != curDims[i]
+      if (firstDims[i] != curDims[i]) {
+        count += 1;
+        index = i;
+      }
+    }
+    if (count > 1) {
+      // If curDims and firstDims have two elements different,
+      // then we can not perform Concat on top of these two nodes.
+      return -1;
+    } else if (count == 0) {
```

Comment:
according to llvm coding standards you should not have `else` when there is a return in the prev condition.

https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Reply:
done